Automattic/newspack-ios
# Issues/58 folder contents core data

This PR is part one of adding a core data layer wrapping folders and files managed by the app.  I'm spacing these out into small chucks to try to make them easier to review, and cut down on what might end up being a lot of boilerplate noise. 

- It contains the changes to the data model and the new managed objects.  
- It also fixes some references to StagedMedia on the Site model that weren't causing issues, but were discovered by regenerating the boilerplate for the Site+CoreDataProperties file.
- It adds an optional (for now) siteFolder property to the Site model.  This will be made non-optional in the future.

Note: I'm not planning to bump the model version until we start cutting alpha releases. Until then I'll just be modifying the single model version as is. 

To test:
- Run tests and ensure everything passes.
- Run the app in the simulator and ensure it builds. 
- Just have a look over the models and make sure nothing looks wonky? 

@jleandroperez Can I trouble you with this one?
